### PR TITLE
Extract dependencies to separate gradle file

### PR DESCRIPTION
**Pull request by ataulm**
Pulls out all the hardcoded versions/dependencies.

This will make it easier to:

- reference dependencies in various modules
- update a version in one place, and have it synced for all places

I didn't update any versions in this PR.

I added a note onto the `targetSdkVersion` - I thought I saw something from @tasomaniac that suggests we shouldn't include this in libraries, but can't remember. This can be amended in this PR.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -15,12 +15,12 @@ apply plugin: 'com.android.library'
...
-    compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    compileSdkVersion androidCompileSdkVersion
+    buildToolsVersion androidBuildToolsVersion
 
     defaultConfig {
-        minSdkVersion 14
-        targetSdkVersion 23
+        minSdkVersion androidMinSdkVersion
+        targetSdkVersion androidTargetSdkVersion
```
Comment (ataulm) [inline android/build.gradle]: @tasomaniac I thought I saw something recently where you mentioned we shouldn't be specifying this - can't find the reference though so not sure if I imagined.

Comment (tasomaniac) [inline android/build.gradle]: Yep, we shouldn't. Here is the change done in Dagger by @JakeWharton in https://github.com/google/dagger/pull/591

When you specify this happens:
- if it is less then the client app's target, no affect at all
- If it is greater, then it takes affect and change the targetSdk of the client.

In either cases, it is not good.

File: dependencies.gradle [hunk 2]
```diff
@@ -0,0 +1,20 @@
+ext {
+    androidCompileSdkVersion = 23
```
Comment (tasomaniac) [inline dependencies.gradle]: We have these inside `versions` array in our project. 
Much similar to this. https://github.com/mr-archano/Playground/blob/develop/gradle/dependencies.gradle

I really like it.

File: java/build.gradle [hunk 3]
```diff
@@ -5,10 +5,11 @@ repositories {
 apply plugin: 'java'
 apply plugin: 'bintray-release'
 
+sourceCompatibility = 1.7
+targetCompatibility = 1.7
+
 dependencies {
-    testCompile 'junit:junit:4.12'
-    sourceCompatibility = 1.7
```
Comment (tasomaniac) [inline java/build.gradle]: haha :)

File: android/build.gradle (hunk 1 above)
Comment (ataulm) [inline android/build.gradle]: Thanks for the link!

commit a4cfae461086874ff079f52ce35968c11e136bca
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Mar 7 22:38:32 2017 +0000

    Remove specification of targetSdk property for library module
    
    - https://github.com/novoda/notils/pull/76#discussion_r104643557

File: dependencies.gradle (hunk 2 above)
Comment (ataulm) [inline dependencies.gradle]: commit 88b11a64abda351aec76b595ec9f13f1d8165675
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Mar 7 22:37:15 2017 +0000

    Extract versions extra properties

File: dependencies.gradle [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        targetCompatibility: JavaVersion.VERSION_1_7,
+        support            : '23.1.1',
+        androidSdk         : [
+            compile   : 23,
+            min       : 14,
+            buildTools: '23.0.2'
+        ]
+    ]
+
+    librariesBuild = [
```
Comment (mr-archano) [inline dependencies.gradle]: what about having 
```
libraries = [
   build = [ ... ],
   app = [ ... ],
   test = [ ... ]
]
```

**Comment by ataulm** [comment]
Good shout on moving it to the root build script!

**Comment by tasomaniac** [comment]
Looks awesome now.

**Comment by zegnus** [comment]
so much improvements, like it! :)

**Pull request merged**